File: Python/Hardware.py
from Product import Product
# mengimport kelas Product


class Hardware(Product):
    # kelas yang digunakan untuk mengimplementasikan kelas Hardware dan juga
    # turunan dari kelas product
    def __init__(self):
        # konstruktor
        self.brand = ""
        self.model = ""

    def setBrand(self, brand):
        # mengeset nilai atribut brand
        self.brand = brand

    def getBrand(self):
        # mengembalikan nilai atribut brand
        return self.brand

    def setModel(self, model):
        # mengeset nilai atribut model
        self.model = model

    def getModel(self):
        # mengembalikan nilai atribut model
        return self.model

    def printHardware(self):
        # menampilkan atribut hardware
        print("Brand: " + self.getBrand())
        print("Model: " + self.getModel())

File: PHP/Hardware.php
<?php
include("Product.php");
# import class product

class Hardware extends Product
# kelas yang digunakan untuk mengimplementasikan Hardware dan juga turunan dari kelas Product
{
    # private atribute dari kelas Hardware
    private $brand = "";
    private $model = "";

    # konstrutor
    public function __construct()
    {
    }

    # mengeset nilai atribut brand
    public function setBrand($brand)
    {
        $this->brand = $brand;
    }

    # mengembalikan nilai atribut brand
    public function getBrand()
    {
        return $this->brand;
    }

    # mengeset nilai atribut model
    public function setModel($model)
    {
        $this->model = $model;
    }

    # mengembalikan nilai atribut model
    public function getModel()
    {
        return $this->model;
    }

    # menampilkan atribut Hardware
    public function printHardware()
    {
        echo "Model: " . $this->getModel() . "<br/>";
        echo "Brand: " . $this->getBrand() . "<br/>";
    }
}
